libtextsecure/message_receiver.js

```diff
@@ -189,12 +195,33 @@ MessageReceiver.prototype.extend({
     }
 
     const promise = Promise.resolve(request.body.toArrayBuffer()) // textsecure.crypto
-      .then(plaintext => {
+      .then(plaintextImmutable => {
+        let plaintext = plaintextImmutable;
+
         const envelope = textsecure.protobuf.Envelope.decode(plaintext);
         // After this point, decoding errors are not the server's
         //   fault, and we should handle them gracefully and tell the
         //   user they received an invalid message
 
+        // The message is for a medium size group
+        if (options.conversationId) {
+          const ourNumber = textsecure.storage.user.getNumber();
+          const senderIdentity = envelope.source;
+
+          if (senderIdentity === ourNumber) {
+            // Ignoring our own message
+            return request.respond(200, 'OK');
+          }
```

Comment:
for some server calls, we reply only when the message is processed really. I think it's a timing security stuff. Shouldn't we do the same here?

Reply:
It doesn't actually do anything at the moment:
```
this.respond = (status, message) => {
  // Mock websocket response
  window.log.info(status, message);
};
```